packages/nhost-js/src/clients/graphql.ts

```diff
@@ -151,15 +103,26 @@ export class NhostGraphqlClient {
    * @docs https://docs.nhost.io/reference/javascript/nhost-js/graphql/set-access-token
    */
   setAccessToken(accessToken: string | undefined) {
-    if (!accessToken) {
-      this.accessToken = null
-      return
-    }
+    this.accessToken = accessToken || null
+    this.resetHeaders()
+  }
+
+  setEndpoint(value: string): GraphQLClient {
```

Comment:
if this is a setter, should it return a `GraphQlClient`?

Reply:
no, that is not required. Setters in js usually return `void`

Reply:
exactly, that is what I mean. We are returning `this` which is the `GraphQlClient`.